certora/specs/Timelock.spec

```diff
@@ -13,13 +13,21 @@ function summarySupplyshares(MetaMorphoHarness.Id id, address user) returns uint
     return res;
 }
 
+persistent ghost uint256 lastTimestamp;
+
+hook TIMESTAMP uint newTimestamp {
```

Comment:
I get the concept, but when exactly is that "executed"? Because in the EVM, the timestamp is never changed, it's more an externality and the evm then realises that it changed

Reply:
Yes it's different from the EVM in the sense that this `lastTimestamp` variable tracks all interactions, including view function calls that could be seen as just looking at the chain. Essentially, it allows to have a notion of the "current time", which you can compare to previous times.
For example, if a function call stores the `block.timestamp`, then you know that the stored value is smaller than the current time. This means that for every interactions that happen after (in the CVL spec, modeling interactions that happen after in the real life) will have a `block.timestamp` that is greater, including for view functions